tests/python/binary_stream/main.py

```diff
@@ -0,0 +1,59 @@
+"""
+Test showing that a binary stream as a valid RRD files.
+
+After running:
+```bash
+rerun *.rrd
+```
+"""
+
+from __future__ import annotations
+
+import queue
+import threading
+import time
+from typing import Any, Iterator
+
+import rerun as rr
+import rerun.blueprint as rrb
+
+
+@rr.thread_local_stream("rerun_example_binary_stream")
+def job(name: str) -> Iterator[tuple[str, bytes]]:
+    stream = rr.binary_stream()
+
+    blueprint = rrb.Blueprint(rrb.TextLogView(name="My Logs", origin="test"))
+
+    rr.send_blueprint(blueprint)
+
+    for i in range(5):
+        time.sleep(0.2)
+        rr.log("test", rr.TextLog(f"Job {name} Message {i}"))
+
+        print(f"YIELD {name} {i}")
+        yield (name, stream.read())
+
+
+def queue_results(generator: Iterator[Any], out_queue: queue.Queue) -> None:
+    for item in generator:
+        out_queue.put(item)
+
+
+if __name__ == "__main__":
+    results_queue: queue.Queue[tuple[str, bytes]] = queue.Queue()
+
+    threads = [
+        threading.Thread(target=queue_results, args=(job("A"), results_queue)),
+        threading.Thread(target=queue_results, args=(job("B"), results_queue)),
+        threading.Thread(target=queue_results, args=(job("C"), results_queue)),
+    ]
+    for t in threads:
+        t.start()
+    for t in threads:
+        t.join()
+
+    while not results_queue.empty():
+        name, data = results_queue.get()
+
+        with open(f"output_{name}.rrd", "a+b") as f:
+            f.write(data)
```

Comment:
if you were to order the output of the queue and write a second rrd file with the same data in sync, then you could binary compare those two files, thus creating an automated test that we can put on ci.
This is too close to that to not do it!

Reply:
Kind of annoying but I'll try to take a shot.

The rrd binary streams are never directly comparable due to randomness in log_time and tow tuid, which means I still need to write out to file and then shell out to rerun_compare.

If we had a python API for retrieving messages back from a stream I totally would have made it a proper unit-test from the start.